cmd/buildkitd/main.go

```diff
@@ -544,27 +544,19 @@ func setDefaultConfig(cfg *config.Config) {
 	}
 }
 
-var (
-	isRootlessConfigOnce  sync.Once
-	isRootlessConfigValue bool
-)
-
 // isRootlessConfig is true if we should be using the rootless config
 // defaults instead of the normal defaults.
-func isRootlessConfig() bool {
-	isRootlessConfigOnce.Do(func() {
-		if !userns.RunningInUserNS() {
-			// Default value is false so keep it that way.
-			return
-		}
-		// if buildkitd is being executed as the mapped-root (not only EUID==0 but also $USER==root)
-		// in a user namespace, we don't want to load the rootless changes in the
-		// configuration.
-		u := os.Getenv("USER")
-		isRootlessConfigValue = u != "" && u != "root"
-	})
-	return isRootlessConfigValue
-}
+var isRootlessConfig = sync.OnceValue(func() bool {
+	if !userns.RunningInUserNS() {
```

Comment:
May be redundant at all; I just recalled it's already using a onceValue; https://github.com/moby/buildkit/blob/ca257713ac550d31edaf6dfc17c37e02c798eedd/vendor/github.com/moby/sys/userns/userns.go#L14-L16

https://github.com/moby/buildkit/blob/ca257713ac550d31edaf6dfc17c37e02c798eedd/vendor/github.com/moby/sys/userns/userns_linux.go#L10

Reply:
Updated to just remove the sync entirely just to simplify things because this is already there.